src/esbmc/esbmc_parseoptions.cpp

```diff
@@ -1648,10 +1648,16 @@ bool esbmc_parseoptionst::process_goto_program(
     namespacet ns(context);
 
     // Start by removing all no-op instructions and unreachable code
-    if(!cmdline.isset("no-remove-no-op"))
+    // We should skip this removal in goto-cov and multi-property
```

Comment:
Because multi-property wants to find all the bugs in the src code. But these optimisations will remove the rest of the code during the Goto stage. e.g.

```c++
int main()

{

   int x[2];
   x[3] = 10;
   switch (nondet_int())
   {
   case 1:
      assert(0);
   case 2:
      assert(0);
   default:
       ;
   }
   return 0;
}
```

will only show one counterexample in multi-property mode:
```
[Counterexample]


State 1 file 2.c line 1251 column 4 function main thread 0
----------------------------------------------------
Violated property:
  file 2.c line 1251 column 4 function main
  array bounds violated: array `x' upper bound
  0


VERIFICATION FAILED
```

Reply:
Also, the idea of goto-coverage is to find out unreached codes(i.e. asserts). Thus we cannot remove the unreachable code beforehand.